src/useCases/getUserByUsername.ts

```diff
@@ -23,28 +26,37 @@ export default async (db: Database, username: string): PromiseResult<User | null
     `
 
   try {
-    const user = await db.oneOrNone<User>(getUserQuery, { username }).catch((error) => error)
+    user = await connection.oneOrNone<User>(getUserQuery, { username }).catch((error) => error)
 
     if (user === null) {
       return null
     }
-
-    return {
-      id: user.id,
-      username: user.username,
-      emailAddress: user.email,
-      exclusionList: user.exclusion_list,
-      inclusionList: user.inclusion_list,
-      endorsedBy: user.endorsed_by,
-      orgServes: user.org_serves,
-      forenames: user.forenames,
-      surname: user.surname,
-      groupId: user.group_id,
-      visibleCourts: user.visible_courts,
-      visibleForces: user.visible_forces,
-      excludedTriggers: user.excluded_triggers
-    }
   } catch (error) {
     return error as Error
   }
+
+  if (user === null) {
```

Comment:
we have already checked if user is null at line 31, no?

Reply:
It seems that way doesn't it

Reply:
Commit comment : 1